### PR TITLE
Back log alban nov

### PR DESCRIPTION
**Pull request by Alban-de-tourtier**
### 🛠 Changes being made

Here give examples of the changes you've made in this pull request. Include an itemized list if you can. It'll help the reviewer

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        {
+            label: 3
+        },
+        {
+            label: 4
+        },
+        {
+            label: 5
+        }
+        ]
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:52]: semicolon missing

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+         getRatingData(restoName)
+             .then(res => setReview(res));
+    };
+    const handleCommentChange = (text) => {
+        setNewComment(text);
+    };
+
+    useEffect(() => {
+        const fetchUserData = async () => {
+                const userToken = await AsyncStorage.getItem('user');
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:70]: userToken instead of user

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    const handleInputChange = (field: keyof IRequestUser, text: string) => {
+        setRequest(prevState => ({
+            ...prevState,
+            [field]: text,
+        }));
+    };
+
+    const getPremium = async () => {
+        try {
+            const userToken = await AsyncStorage.getItem('user');
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:91]: userToken instead of user and null check missing

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        <View style={[styles.container]}>
+            <Text style={styles.mainTitle}>{t('pages.Review.add-review')}</Text>
+            <Text>{t('pages.Review.your-review')}</Text>
+            <Dropdown
+                style={styles.dropdown}
+                data={note}
+                maxHeight={300}
+                labelField="label"
+                valueField="label"
+                placeholder={'Note'}
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:138]: translate here and in line below

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                value={value}
+                onChange={(note) => {
+                    setValue(note.label);
+                }}
+            />
+            <TextInput
+                style={styles.input}
+                onChangeText={handleCommentChange}
+                value={newComment}
+                placeholder={"Your comment"}
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:149]: translate

File: visitorMobile/src/pages/UserReviewPage/UserReview.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+const RatingPage: React.FC<ReviewPageProps> = ({route}) => {
+    const [request, setRequest] = useState<IRequestUser>(initialRequestState);
+    const {t} = useTranslation();
+    const [darkMode, setDarkMode] = useState<boolean>(false);
+    const [name, setName] = useState<string>();
+    const [userReview, setUserReview] = useState([])
+
+    useEffect(() => {
+        const fetchUserData = async () => {
+                const userToken = await AsyncStorage.getItem('user');
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/UserReviewPage/UserReview.tsx:32]: 'userToken'

File: visitorMobile/src/pages/UserReviewPage/UserReview.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    const handleInputChange = (field: keyof IRequestUser, text: string) => {
+        setRequest(prevState => ({
+            ...prevState,
+            [field]: text,
+        }));
+    };
+
+    const getPremium = async () => {
+        try {
+            const userToken = await AsyncStorage.getItem('user');
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/UserReviewPage/UserReview.tsx:59]: userToken
null check missing

File: visitorMobile/src/services/ratingCalls.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+
+// @ts-ignore
+import { API_URL } from '@env';
+
+const baseUrl =
+    `${API_URL}`;
```
Comment (JosefineM2000) [inline visitorMobile/src/services/ratingCalls.ts:7]: newline not needed

File: visitorMobile/src/pages/UserReviewPage/UserReview.tsx
Comment (JosefineM2000) [inline visitorMobile/src/pages/UserReviewPage/UserReview.tsx:1]: Isn't that the same as Rating.tsx?

File: visitorMobile/src/components/RestaurantCard/RestaurantCard.tsx [hunk 9]
```diff
@@ -122,6 +153,9 @@ const RestaurantCard = (props: RestaurantCardProps) => {
               />
             </TouchableOpacity>
           </Text>
+          <TouchableOpacity  style={styles.button} onPress={navigateToReview}>
```
Comment (JosefineM2000) [inline visitorMobile/src/components/RestaurantCard/RestaurantCard.tsx:156]: I cant click on that, it leads me to the menu

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

File: visitorMobile/src/pages/UserReviewPage/UserReview.tsx
Comment (Alban-de-tourtier) [reply visitorMobile/src/pages/UserReviewPage/UserReview.tsx:1]: no you can delete and you only see your review

File: visitorMobile/src/components/RestaurantCard/RestaurantCard.tsx (hunk 9 above)
Comment (Alban-de-tourtier) [reply visitorMobile/src/components/RestaurantCard/RestaurantCard.tsx:156]: can I have a screenrecord?

File: visitorMobile/src/pages/RatingPage/Rating.tsx [hunk 10]
```diff
@@ -51,6 +53,11 @@ const RatingPage: React.FC<ReviewPageProps> = ({route}) => {
         }
     ];
      const addReview = async () => {
+         if (newComment === undefined || value === null) {
+             setInvalidInput(true)
+             return
+         } else {
+         }
```
Comment (JosefineM2000) [inline visitorMobile/src/pages/RatingPage/Rating.tsx:60]: else what?

Comment (MarcPIster) [reply visitorMobile/src/pages/RatingPage/Rating.tsx:48]: fixed

**Review by MarcPIster — APPROVED** [review]

**Review by JosefineM2000 — APPROVED** [review]
